### PR TITLE
feat(storage): avoids multiple offers creation

### LINKED ISSUE #526: Prevent multiple offers creation
If the provider already has an offer and creates a new one on `Storage/Sell` the reality is that the previous offer is being edited.

We should prevent this situation and probably tell the user to edit the offer on `Storage/My Offers`.

### PR DESCRIPTION
**Pull request by farinalvaro**
Closes #526

### CODE REVIEW

File: src/components/pages/storage/sell/Page.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      : (
+        <OfferEditContextProvider>
+          <OfferCreation isLoading={isLoadingOffer} />
+        </OfferEditContextProvider>
+      )
+  )
+
+  return (
+    <CenteredPageTemplate>
+      {renderContent()}
```
Comment (jurajpiar) [inline src/components/pages/storage/sell/Page.tsx:58]: seems unnecessary. just put the content here no?

File: src/components/pages/storage/sell/Page.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          provider: account,
+        }) as StorageOffer[]
+
+        setOwnOffer(currentOwnOffers[0])
+        setIsLoadingOffer(false)
+      }
+
+      getOwnOffers()
+    }
+  }, [account, appDispatch, appState])
```
Comment (jurajpiar) [inline src/components/pages/storage/sell/Page.tsx:42]: I'd prefer you put this in the context provider.

File: src/components/pages/storage/sell/NoMultipleOffers.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+              </Box>
+            </NavLink>
+            {' section.'}
+          </Typography>
+        </Grid>
+        <Grid container justify="center">
+          <RoundBtn onClick={navigateMyOffers}>Go to My Offers</RoundBtn>
+        </Grid>
+      </TitledRoundedCard>
+    </Grid>
```
Comment (jurajpiar) [inline src/components/pages/storage/sell/NoMultipleOffers.tsx:55]: Not sure the extraction is needed. Consider rendering it directly in the sell page. My argument is that if you take the out the stuff that should be in the context the component wont get much bigger by having this code in.

File: src/components/pages/storage/sell/NoMultipleOffers.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+const useStyles = makeStyles(() => ({
+  myOffersLink: {
+    color: colors.gray4,
+  },
+}))
+
+const MY_OFFERS_ROUTE = ROUTES.STORAGE.MYOFFERS.BASE
+
+const NoMultipleOffers: FC = () => {
```
Comment (jurajpiar) [inline src/components/pages/storage/sell/NoMultipleOffers.tsx:20]: If you are going to keep this extraction, as I suspect you will, then rename it to something more telling of the job of this component. Basically it needs a noun in the name. And although "offers" IS as noun, in this combination it acts more as an adjective. So, NoMultipleOffers ... what? I'd say NoMultipleOffersCard would be the minimal effort acceptable change.

**Review by jurajpiar — CHANGES_REQUESTED** [review]
Please consider the comments...

File: src/components/pages/storage/sell/Page.tsx (hunk 2 above)
Comment (farinalvaro) [reply src/components/pages/storage/sell/Page.tsx:42]: There is no context provider for this page and I think it's its responsibility to know if there's an offer created and so act based on that.
To me it seems like creating a context only to fetch an offer would add more complexity. I just wanted to keep it simple

File: src/components/pages/storage/sell/Page.tsx (hunk 1 above)
Comment (farinalvaro) [reply src/components/pages/storage/sell/Page.tsx:58]: I just thought it was cleaner as this is a conditional component. Let me know if you still want me to move this.

File: src/components/pages/storage/sell/NoMultipleOffers.tsx (hunk 3 above)
Comment (farinalvaro) [reply src/components/pages/storage/sell/NoMultipleOffers.tsx:55]: What's the disadvantage of having it as a separated component? I find this cleaner and with better performance

Comment (farinalvaro) [reply src/components/pages/storage/sell/NoMultipleOffers.tsx:55]: Resolving after our chat

File: src/components/pages/storage/sell/NoMultipleOffers.tsx (hunk 4 above)
Comment (farinalvaro) [reply src/components/pages/storage/sell/NoMultipleOffers.tsx:20]: Resolving after our chat

File: src/components/pages/storage/sell/Page.tsx (hunk 2 above)
Comment (farinalvaro) [reply src/components/pages/storage/sell/Page.tsx:42]: Resolving after our chat

File: src/components/pages/storage/sell/NoMultipleOffersPage.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        roundedCardProps={{ color: 'secondary' }}
+      >
+        <Grid item xs={12}>
+          <Typography
+            component="div"
+            color="secondary"
+            align="center"
+            gutterBottom
+          >
+            Creation of multiple offers is not supported yet.
```
Comment (julianmrodri) [inline src/components/pages/storage/sell/NoMultipleOffersPage.tsx:40]: I would change the message to a more positive like: "An offer is already created for the selected account."

File: src/components/pages/storage/sell/NoMultipleOffersPage.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            <br />
+            {'You can edit your current offer on '}
```
Comment (julianmrodri) [inline src/components/pages/storage/sell/NoMultipleOffersPage.tsx:42]:
```suggestion
            {'You can edit your current offer in '}
```

**Review by julianmrodri — COMMENTED** [review]
Texting and wording changes. Besides that is Ok.

File: src/components/pages/storage/sell/Wrapper.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import { Web3Store } from '@rsksmart/rif-ui'
+import AppContext, { errorReporterFactory } from 'context/App/AppContext'
+import { OfferEditContextProvider } from 'context/Market/storage/OfferEditContext'
+import { StorageOffer } from 'models/marketItems/StorageItem'
+import { StorageOffersService } from 'api/rif-marketplace-cache/storage/offers'
+import CenteredPageTemplate from 'components/templates/CenteredPageTemplate'
+import OfferCreationPage from './OfferCreationPage'
+import NoMultipleOffersPage from './NoMultipleOffersPage'
+
+const Wrapper: FC = () => {
```
Comment (jurajpiar) [inline src/components/pages/storage/sell/Wrapper.tsx:13]: consider naming it `OfferPageWrapper` instead

**Review by jurajpiar — COMMENTED** [review]
Please consider the comments...

File: src/components/pages/storage/sell/Wrapper.tsx (hunk 7 above)
Comment (farinalvaro) [reply src/components/pages/storage/sell/Wrapper.tsx:13]: Done

**Review by jurajpiar — DISMISSED** [review]

**Review by jurajpiar — APPROVED** [review]

**Pull request merged**